storage-node/src/disk/disk_manager.rs

```diff
@@ -0,0 +1,69 @@
+use std::fs::{self, File, OpenOptions};
+use std::io::{self, Read, Write};
+use std::path::{Path, PathBuf};
+
+use super::disk_stream::DiskReadSyncIterator;
+
+/// [`DiskManager`] contains the common logic to read from or write to a disk.
+///
+/// FIXME: We should store base path in cache
+/// TODO: add some statistics member in DiskManager
+pub struct DiskManager {}
+
+impl DiskManager {
+    pub fn get_write_file(&self, path: &str, append: bool) -> io::Result<File> {
+        let path_buf: PathBuf = PathBuf::from(path);
+        if let Some(parent) = path_buf.parent() {
+            if !parent.exists() {
+                std::fs::create_dir_all(parent)?;
+            }
+        }
+        OpenOptions::new()
+            .write(true)
+            .append(append)
+            .open(&path_buf)
+    }
+
+    // FIXME: `mut` allows future statistics computation
+    // Do not call this method in a loop, otherwise everytime you have to repoen the file
+    // When you want to write to disk in a loop due to huge size of file, please first
+    // call get_write_file, and use `file.write_all` directly in a loop.
+    // But we need to record statistics... so `disk_manager` should provide this method...
+    // Maybe something like `DiskWriteSyncIterator` is needed... It's weird though
+
+    // TODO: only practical when you want to write data all at once, will be modified
+    // TODO: the ideal situation is directly read from S3, directly write to cache,
+    //       without memory involved, if no memory cache
+    //       so after determining the S3 read func, `disk_manager` can provide such method
```

Comment:
This sounds like some advanced hardware-involved tricks? Or anyway we have to read the bytes into the memory and then write them out. Or do you mean stream-read and stream-write so that we don't keep every piece of the Parquet file in memory at the same time?